### PR TITLE
IGAPP-941: split regex and don't allow . at the end of the string

### PR DESCRIPTION
**Pull request by sarahsporck**
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **IGAPP**.

Tested here https://regex101.com/r/ZfeAMj/1
If we also don't want to allow trailing , . / @ ... it is enough to set \b at the end of the regex. Opinions?
I also split the regex for a better overview. If you disagree with splitting it I can combine it again and keep the split part as a comment.

### CODE REVIEW

File: api-client/src/replaceLinks.ts [hunk 1]
```diff
@@ -1,6 +1,11 @@
 // https://stackoverflow.com/a/3809435
-const regex =
-  /((([A-Za-z]{3,9}:(?:\/\/)?)(?:[-;:&=+$,\w]+@)?[A-Za-z0-9.-]+|[-;:&=+$,\w]+@[A-Za-z0-9.-]+)((?:\/[+~%/.\w\-_]*)?\??(?:[-+=&;%@.\w_]*)#?(?:[.!/\\\w]*))?)/g
+const protocol = /([A-Za-z]{3,9}:(?:\/\/)?)/
+const baseUrl =  /(?:[-;:&=+$,\w]+@)?([A-Za-z0-9-]+(?:\.[A-Za-z0-9-]+)*)/
```
Comment (klinzo) [inline api-client/src/replaceLinks.ts:3]:
```suggestion
const hostname =  /(?:[-;:&=+$,\w]+@)?([A-Za-z0-9-]+(?:\.[A-Za-z0-9-]+)*)/
```

**Review by klinzo — APPROVED** [review]
Nice :+1: I like the splitting. I also think it makes sense to match word boundaries (\b) to avoid matching of other unwanted trailing characters.

**Comment by sarahsporck** [comment]
> Nice :+1: I like the splitting. I also think it makes sense to match word boundaries (\b) to avoid matching of other unwanted trailing characters.

But keep in mond this includes the trailing slash, which was the main reason why I didn't use it right away.

**Review by SteffiStuffel — APPROVED** [review]
Tested on web: Dot at the end of the link is still displayed as part of the link, but works fine and opens url without link: http://localhost:9000/testumgebung/de/bilder

**Comment by klinzo** [comment]
> But keep in mond this includes the trailing slash, which was the main reason why I didn't use it right away.

Okay yes, missed that. Then we should not do that, you are right. But maybe it makes sense to not match at least commas `,`.

**Pull request merged**